src/Perlang.Interpreter/Compiler/PerlangCompiler.cs

```diff
@@ -490,28 +490,49 @@ private void RegisterGlobalClasses()
                 }
             }
 
-            // TODO: Try picking this up from the location of the `perlang` executable first, if possible. We might even
-            // TODO: be able to get rid of this environment variable completely.
-            string? perlangRoot = Environment.GetEnvironmentVariable("PERLANG_ROOT");
+            // We attempt to detect the stdlib location by looking in the path to the `perlang` executable first. If it
+            // doesn't exist there, the PERLANG_ROOT environment variable is checked next.
+            string? executablePath = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
 
-            if (perlangRoot == null)
+            string? stdlibPath;
+
+            string perlangExecutableStdlibPath = Path.Join(executablePath, "lib", "stdlib");
+
+            if (Directory.Exists(perlangExecutableStdlibPath))
             {
-                throw new PerlangCompilerException(
-                    "The PERLANG_ROOT environment variable must be set for experimental compilation to succeed. " +
-                    "It should point to a directory where the lib/stdlib directory contains a compiled version " +
-                    "of the Perlang standard library.");
+                // This seems to be a Perlang distribution, either a nightly snapshot or a release version. Use the
+                // stdlib provided with it.
+                stdlibPath = perlangExecutableStdlibPath;
+            }
+            else
+            {
+                string? perlangRoot = Environment.GetEnvironmentVariable("PERLANG_ROOT");
+
+                if (perlangRoot == null)
+                {
+                    throw new PerlangCompilerException(
+                        $"The {perlangExecutableStdlibPath} directory does not exist, and PERLANG_ROOT is not set. One " +
+                        "of these conditions must be met for experimental compilation to succeed. If setting " +
+                        "PERLANG_ROOT, it should point to a directory where the lib/stdlib directory contains a " +
+                        "compiled version of the Perlang standard library."
+                    );
+                }
+
+                stdlibPath = Path.Join(perlangRoot, "lib", "stdlib");
             }
 
             var processStartInfo = new ProcessStartInfo
             {
-                FileName = "clang++",
+                // Make this explicit, since we have only tested this with a very specific clang version. Anything else is
+                // completely untested and not expected to work at the moment.
+                FileName = "clang++-14",
```

Comment:
I deliberately made this a bit more strict. Let's see what error we get if you try to run this in a container without any `clang-14` package installed btw...

Reply:
Here we go. With an added `catch (SystemException e)`, and re-throwing this, we get something which is good enough for now:

```
root@ce9d3c51d947:/perlang# /home/per/.perlang/nightly/bin/perlang docs/examples/quickstart/pi.per
Unhandled exception. System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.ApplicationException: Failed running clang. Experimental compilation is only supported on Linux-based systems. If running a Debian/Ubuntu-based distribution, please ensure that the clang-14 package is installed since experimental compilation depends on it.
 ---> System.ComponentModel.Win32Exception (2): An error occurred trying to start process 'clang++-14' with working directory '/perlang'. No such file or directory
   at System.Diagnostics.Process.ForkAndExecProcess(ProcessStartInfo, String, String[], String[], String, Boolean, UInt32, UInt32, UInt32[], Int32& , Int32& , Int32& , Boolean, Boolean )
   at System.Diagnostics.Process.StartCore(ProcessStartInfo)
   at System.Diagnostics.Process.Start()
   at System.Diagnostics.Process.Start(ProcessStartInfo)
   at Perlang.Interpreter.Compiler.PerlangCompiler.Compile(String source, String path, CompilerFlags compilerFlags, ScanErrorHandler scanErrorHandler, ParseErrorHandler parseErrorHandler, NameResolutionErrorHandler nameResolutionErrorHandler, ValidationErrorHandler typeValidationErrorHandler, ValidationErrorHandler immutabilityValidationErrorHandler, CompilerWarningHandler compilerWarningHandler) in /home/per/git/perlang/src/Perlang.Interpreter/Compiler/PerlangCompiler.cs:line 607
   --- End of inner exception stack trace ---
   at Perlang.Interpreter.Compiler.PerlangCompiler.Compile(String source, String path, CompilerFlags compilerFlags, ScanErrorHandler scanErrorHandler, ParseErrorHandler parseErrorHandler, NameResolutionErrorHandler nameResolutionErrorHandler, ValidationErrorHandler typeValidationErrorHandler, ValidationErrorHandler immutabilityValidationErrorHandler, CompilerWarningHandler compilerWarningHandler) in /home/per/git/perlang/src/Perlang.Interpreter/Compiler/PerlangCompiler.cs:line 632
   at Perlang.Interpreter.Compiler.PerlangCompiler.CompileAndRun(String source, String path, CompilerFlags compilerFlags, ScanErrorHandler scanErrorHandler, ParseErrorHandler parseErrorHandler, NameResolutionErrorHandler nameResolutionErrorHandler, ValidationErrorHandler typeValidationErrorHandler, ValidationErrorHandler immutabilityValidationErrorHandler, CompilerWarningHandler compilerWarningHandler) in /home/per/git/perlang/src/Perlang.Interpreter/Compiler/PerlangCompiler.cs:line 221
   at Perlang.ConsoleApp.Program.CompileAndRun(String source, String path, CompilerWarningHandler compilerWarningHandler) in /home/per/git/perlang/src/Perlang.ConsoleApp/Program.cs:line 456
   at Perlang.ConsoleApp.Program.RunFile(String path) in /home/per/git/perlang/src/Perlang.ConsoleApp/Program.cs:line 344
   at Perlang.ConsoleApp.Program.<>c__DisplayClass13_0.<MainWithCustomConsole>b__4(ParseResult parseResult, IConsole _) in /home/per/git/perlang/src/Perlang.ConsoleApp/Program.cs:line 249
   at System.RuntimeMethodHandle.InvokeMethod(Object, Void**, Signature, Boolean)
   at System.Reflection.MethodInvoker.Invoke(Object, IntPtr*, BindingFlags)
   --- End of inner exception stack trace ---
   at System.Reflection.MethodInvoker.Invoke(Object, IntPtr*, BindingFlags)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object, BindingFlags, Binder, Object[], CultureInfo)
   at System.Delegate.DynamicInvokeImpl(Object[] args)
   at System.CommandLine.NamingConventionBinder.ModelBindingCommandHandler.InvokeAsync(InvocationContext context)
   at System.CommandLine.NamingConventionBinder.ModelBindingCommandHandler.Invoke(InvocationContext context)
   at System.CommandLine.Invocation.InvocationPipeline.<>c__DisplayClass4_0.<<BuildInvocationChain>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass12_0.<<UseHelp>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Invocation.InvocationPipeline.<Invoke>g__FullInvocationChain|3_0(InvocationContext)
   at Perlang.ConsoleApp.Program.MainWithCustomConsole(String[] args, IPerlangConsole console) in /home/per/git/perlang/src/Perlang.ConsoleApp/Program.cs:line 292
   at Perlang.ConsoleApp.Program.Main(String[] args) in /home/per/git/perlang/src/Perlang.ConsoleApp/Program.cs:line 108
Aborted (core dumped)
```